### PR TITLE
Remove mentions of attr from docs

### PR DESCRIPTION
**Pull request by jl-wynen**
Part of #3166.

I did not yet touch the data creation scripts ~`prepare_data_rhessi.py` and `prepare-filtering-data.py`~. The former should wait until #3258 is merged ~and I haven't looked in detail at the latter yet~.

Also, I did not touch ADRs after talking to Neil about it. We agreed that ADRs should not change after they were accepted.

EDIT
Migrated the event filtering example.

EDIT
Migrated the solar flares example.
Unfortunately, NASA updated the flare list and it is now filtered such that the tutorial doesn't make much sense. So I stuck with the old data. But since I don't have a copy of the old flare list, I modified our old hdf5 file.

### CODE REVIEW

**Review by nvaytet — COMMENTED** [review]
Looks good overall. Some small comments. I have to say I did not go through the scipp source to see if any other mentions of attributes were missed, I only checked that what is in this PR looks ok.

File: docs/tutorials/solar_flares.ipynb [hunk 1]
```diff
@@ -138,9 +129,35 @@
    "outputs": [],
    "source": [
     "filename = sc.data.rhessi_flares()\n",
-    "flares = sc.io.load_hdf5(filename)\n",
-    "# Move min_energy from attributes to coordinates\n",
-    "flares.coords['min_energy'] = flares.attrs.pop('min_energy')\n",
+    "flare_data = sc.io.load_hdf5(filename)\n",
+    "flare_data"
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/tutorials/solar_flares.ipynb:133]: I always struggle to come up with good names in this situation. It's similar to loading nexus files:
```python
<needs-a-name> = load()
detector = <needs-a-name>['instrument']['detector']
```
Do you have a good idea what to use for `<needs-a-name>`?

File: docs/user-guide/slicing.ipynb [hunk 2]
```diff
@@ -240,14 +236,14 @@
     "\n",
     "When slicing a data array the following additional rule applies:\n",
     "\n",
-    "- Meta data (coords, masks, attrs) that *do not depend on the slice dimension* are marked as *readonly*\n",
+    "- Meta data (coords, masks) that *do not depend on the slice dimension* are marked as *readonly*\n",
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/user-guide/slicing.ipynb:239]: According to my grammar checker, Wikipedia, and Merriam-Webster it's 'metadata'.
It's inconsistent in our docs.

File: docs/tutorials/solar_flares.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvaytet) [reply docs/tutorials/solar_flares.ipynb]: > Do you have a good idea what to use for `<needs-a-name>`?

No.

I see in this PR you changed to `flare_datagroup`, which is fine. I guess you could also have used something like `flare_database`, if you wanted to stay away from the Scipp types, but either is fine.

**Review by nvaytet — APPROVED** [review]
Looks good but there are merge conflicts....

**Pull request merged**